postgre-client/src/asset_index_client.rs

```diff
@@ -244,92 +269,204 @@ pub(crate) fn split_assets_into_components(asset_indexes: &[AssetIndex]) -> Asse
+        let dump_result = match asset_type {
+            AssetType::NonFungible => {
+                let Some(metadata_path) =
+                    base_path.join("metadata.csv").to_str().map(str::to_owned)
+                else {
+                    return Err(IndexDbError::BadArgument(format!(
+                        "invalid path '{:?}'",
+                        base_path
+                    )));
+                };
+                let Some(creators_path) =
+                    base_path.join("creators.csv").to_str().map(str::to_owned)
+                else {
+                    return Err(IndexDbError::BadArgument(format!(
+                        "invalid path '{:?}'",
+                        base_path
+                    )));
+                };
+                let Some(assets_authorities_path) = base_path
+                    .join("assets_authorities.csv")
+                    .to_str()
+                    .map(str::to_owned)
+                else {
+                    return Err(IndexDbError::BadArgument(format!(
+                        "invalid path '{:?}'",
+                        base_path
+                    )));
+                };
+                let Some(assets_path) = base_path.join("assets.csv").to_str().map(str::to_owned)
+                else {
+                    return Err(IndexDbError::BadArgument(format!(
+                        "invalid path '{:?}'",
+                        base_path
+                    )));
+                };
+
+                self.copy_nfts(
+                    metadata_path,
+                    creators_path,
+                    assets_path,
+                    assets_authorities_path,
+                    &mut transaction,
+                )
+                .await?;
 
-        self.copy_all(
-            metadata_path,
-            creators_path,
-            assets_path,
-            assets_authorities_path,
-            fungible_tokens_path,
-            &mut transaction,
-        )
-        .await?;
-        self.update_last_synced_key(last_key, &mut transaction, "last_synced_key")
-            .await?;
-        self.commit_transaction(transaction).await?;
-        Ok(())
+                self.update_last_synced_key(
+                    last_key,
+                    &mut transaction,
+                    "last_synced_key",
+                    asset_type,
+                )
+                .await
+            }
+            AssetType::Fungible => {
+                let Some(fungible_tokens_path) = base_path
+                    .join("fungible_tokens.csv")
+                    .to_str()
+                    .map(str::to_owned)
+                else {
+                    return Err(IndexDbError::BadArgument(format!(
+                        "invalid path '{:?}'",
+                        base_path
+                    )));
+                };
+
+                self.copy_fungibles(fungible_tokens_path, &mut transaction)
+                    .await?;
```

Comment:
The explicit rollback will not be called here, if the copy_fungible errors, so we'll still rely on rust to rollback the tx